jetty-servlet/src/main/java/org/eclipse/jetty/servlet/StatisticsServlet.java

```diff
@@ -80,47 +88,150 @@ public void init() throws ServletException
     }
 
     @Override
-    public void doPost(HttpServletRequest sreq, HttpServletResponse sres) throws ServletException, IOException
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
     {
-        doGet(sreq, sres);
+        doGet(request, response);
     }
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
     {
         if (_statsHandler == null)
         {
             LOG.warn("Statistics Handler not installed!");
-            resp.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
+            response.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
             return;
         }
         if (_restrictToLocalhost)
         {
-            if (!isLoopbackAddress(req.getRemoteAddr()))
+            if (!isLoopbackAddress(request.getRemoteAddr()))
             {
-                resp.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
+                response.sendError(HttpServletResponse.SC_FORBIDDEN);
                 return;
             }
         }
 
-        if (Boolean.parseBoolean(req.getParameter("statsReset")))
+        if (Boolean.parseBoolean(request.getParameter("statsReset")))
         {
+            response.setStatus(HttpServletResponse.SC_OK);
             _statsHandler.statsReset();
             return;
         }
 
-        String wantXml = req.getParameter("xml");
-        if (wantXml == null)
-            wantXml = req.getParameter("XML");
+        List<String> acceptable = getOrderedAcceptableMimeTypes(request);
+
+        for (String mimeType : acceptable)
+        {
+            switch (mimeType)
+            {
+                case "application/json":
+                    writeJsonResponse(response);
+                    return;
+                case "text/xml":
+                    writeXmlResponse(response);
+                    return;
+                case "text/html":
+                    writeHtmlResponse(response);
+                    return;
+                case "text/plain":
+                    writeTextResponse(response);
+                    return;
+                case "*/*":
+                    String wantXml = request.getParameter("xml");
```

Comment:
I think this mechanism is now insufficient.
Instead I think we need `type` parameter that can be `json`,`xml`, `html` or `plain`.
Only if their is no `type` parameter should we look at accepted types
and finally if there is no preference from the accepted types, we should just send plain.... maybe deprecated support for `xml` parameter

Reply:
Pushed a change, is this what you are asking for?

Reply:
I still think there is a need for a parameter as it can be difficult to set the accept header in some clients.

Reply:
> I still think there is a need for a parameter as it can be difficult to set the accept header in some clients.

If that were true, REST wouldn't exist like it does today.  
You can even set the `Accept` header on a browser when for asking for content in various ways.

Reply:
Variation using query parameters incoming ...

Reply:
Updated for variation on `accept` query parameter.

Reply:
@gregw bump, is the new `accept` query parameter ... uhm ... acceptable to you?